### PR TITLE
Added 404 and 500 error pages 

### PR DESCRIPTION
**Pull request by alufers**
I have added a 404 and 500 error page, using the main layout and some troubleshooting tips.

### CODE REVIEW

**Comment by alufers** [comment]
Hmm I don't understand why the old commits are showing up here, since they were merged yesterday...

File: static/css/custom.css [hunk 1]
```diff
@@ -88,20 +88,16 @@ body.fixed-nav {
...
 
-#mainNav .navbar-collapse .navbar-sidenav .sidenav-second-level,
-#mainNav .navbar-collapse .navbar-sidenav .sidenav-third-level {
+#mainNav .navbar-collapse .navbar-sidenav .sidenav-second-level, #mainNav .navbar-collapse .navbar-sidenav .sidenav-third-level {
   padding-left: 0;
 }
 
-#mainNav .navbar-collapse .navbar-sidenav .sidenav-second-level > li > a,
-#mainNav .navbar-collapse .navbar-sidenav .sidenav-third-level > li > a {
+#mainNav .navbar-collapse .navbar-sidenav .sidenav-second-level > li > a, #mainNav .navbar-collapse .navbar-sidenav .sidenav-third-level > li > a {
```
Comment (saurabhshri) [inline static/css/custom.css]: Please undo all these formatting changes :)

File: static/js/pong.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+(function() {
```
Comment (saurabhshri) [inline static/js/pong.js:4]: Did you create this game?

File: templates/404.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+{% block content %}
+{{ super() }}
+<div class="container">
+    <div class="mt-10"></div>
+    <h1>Error 404 - Page not found</h1>
+    <p>You can:
+        <ul>
+            <li>Check the URL for errors and mistakes</li>
+            <li><a href="#" id="playPong">Play pong</a></li>
+            <li>Submit an issue on <a href="https://github.com/saurabhshri/ccextractor-web/issues">GitHub</a></li>
```
Comment (saurabhshri) [inline templates/404.html]: Don't hardcode this link. Maybe you can write something generic like 'get in touch` or fetch the link from a config file.

File: templates/404.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        </ul>
+    </p>
+    <canvas width="1000" height="500" id="pongCanvas">Your browser doesn't support html5 canvas.</canvas>
+</div>
+
```
Comment (saurabhshri) [inline templates/404.html:17]: Please make use of `layout` you're fetching in the handler file. Currently the sidebar has no entries.

File: templates/500.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            <li>Try again later</li>
+            <li><a href="#" id="playPong">Play pong</a></li>
+            <li>Submit an issue on <a href="https://github.com/saurabhshri/ccextractor-web/issues">GitHub</a></li>
+        </ul>
+    </p>
+    <canvas width="1000" height="500" id="pongCanvas">Your browser doesn't support html5 canvas.</canvas>
+</div>
+
+{% endblock %}
+
```
Comment (saurabhshri) [inline templates/500.html:19]: Please make use of layout you're fetching in the handler file. Currently the sidebar has no entries.

File: templates/500.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+{% block content %}
+{{ super() }}
+<div class="container">
+    <div class="mt-10"></div>
+    <h1>Error 500 - Internal server error</h1>
+    <p>You can:
+        <ul>
+            <li>Try again later</li>
+            <li><a href="#" id="playPong">Play pong</a></li>
+            <li>Submit an issue on <a href="https://github.com/saurabhshri/ccextractor-web/issues">GitHub</a></li>
```
Comment (saurabhshri) [inline templates/500.html]: Don't hardcode this link. Maybe you can write something generic like 'get in touch` or fetch the link from a config file.

**Review by saurabhshri — CHANGES_REQUESTED** [review]

**Comment by saurabhshri** [comment]
Thank you for the PR! :) Looks great, just some little changes before we can merge.

File: static/js/pong.js (hunk 2 above)
Comment (alufers) [reply static/js/pong.js:4]: Yes, out of boredom on a flight a few weeks ago.

Comment (saurabhshri) [reply static/js/pong.js:4]: Hah, looks nice! :) Feel free to include your attribution in it if you want to. Also, I tried playing it, and there might be a small bug. The bats can go out of the screen, so maybe set a boundary while you're at it?

File: run.py [hunk 7]
```diff
@@ -228,6 +228,22 @@ def before_first_request():
     os.makedirs(os.path.dirname(os.path.join(app.config['LOGS_DIR'])), exist_ok=True)
     os.makedirs(os.path.dirname(os.path.join(app.config['OUTPUT_DIR'])), exist_ok=True)
 
+@app.errorhandler(404)
+def page_not_found_handler(e):
+    """
+    Handles the 404 error and displays a template.
+    """
+    layout = LayoutHelper(logged_in=False)
```
Comment (saurabhshri) [inline run.py]: A user could be logged in and still encounter a 404 ;)

File: run.py [hunk 8]
```diff
@@ -228,6 +228,22 @@ def before_first_request():
...
+    return render_template('404.html', layout=layout.get_entries()), 404
+
+
+@app.errorhandler(500)
+def internal_server_error_handler(e):
+    """
+    Handles the 500 error and displays a template.
+    """
+    layout = LayoutHelper(logged_in=False)
```
Comment (saurabhshri) [inline run.py]: Same. A user could be logged in and still encounter a 500  ;)

**Review by saurabhshri — CHANGES_REQUESTED** [review]

File: run.py (hunk 7 above)
Comment (alufers) [reply run.py]: Good point, will fix that

File: run.py [hunk 9]
```diff
@@ -228,6 +228,26 @@ def before_first_request():
...
     os.makedirs(os.path.dirname(os.path.join(app.config['OUTPUT_DIR'])), exist_ok=True)
 
+@app.errorhandler(404)
+def page_not_found_handler(e):
+    """
+    Handles the 404 error and displays a template.
+    """
+    if g.user is not None:
+        details = DetailsForTemplate(g.user.id, admin_dashboard=False)
+    layout = LayoutHelper(logged_in=not g.user is None, details=details)
```
Comment (saurabhshri) [inline run.py]: if `g.user` is `None` then `details` will be undefined.

**Review by saurabhshri — CHANGES_REQUESTED** [review]

**Review by saurabhshri — CHANGES_REQUESTED** [review]
Please replace it with

```
    if g.user is not None:
        details = DetailsForTemplate(g.user.id, admin_dashboard=False)
        layout = LayoutHelper(logged_in=True, details=details)

    else:
        layout = LayoutHelper(logged_in=False)
```

**Comment by alufers** [comment]
Done

**Comment by saurabhshri** [comment]
Fantastic! Thank you! :)

**Pull request merged**